rpcserver.go

```diff
@@ -1875,6 +1880,337 @@ func (r *rpcServer) AnchorVirtualPsbts(ctx context.Context,
 	}, nil
 }
 
+// CommitVirtualPsbts creates the output commitments and proofs for the given
+// virtual transactions by committing them to the BTC level anchor transaction.
+// In addition, the BTC level anchor transaction is funded and prepared up to
+// the point where it is ready to be signed.
+func (r *rpcServer) CommitVirtualPsbts(ctx context.Context,
+	req *wrpc.CommitVirtualPsbtsRequest) (*wrpc.CommitVirtualPsbtsResponse,
+	error) {
+
+	if len(req.VirtualPsbts) == 0 {
+		return nil, fmt.Errorf("no virtual PSBTs specified")
+	}
+
+	pkt, err := psbt.NewFromRawBytes(bytes.NewReader(req.AnchorPsbt), false)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding packet: %w", err)
+	}
+
+	vPackets := make([]*tappsbt.VPacket, len(req.VirtualPsbts))
+	for idx := range req.VirtualPsbts {
+		vPackets[idx], err = tappsbt.Decode(req.VirtualPsbts[idx])
+		if err != nil {
+			return nil, fmt.Errorf("error decoding virtual "+
+				"packet at index %d: %w", idx, err)
+		}
+	}
+
+	// Make sure we decorate all asset inputs with the correct internal key
+	// derivation path (if it's indeed a key this daemon owns).
+	for idx := range pkt.Inputs {
+		pIn := &pkt.Inputs[idx]
+
+		// We only care about asset inputs which always specify a
+		// Taproot merkle root.
+		if len(pIn.TaprootMerkleRoot) == 0 {
```

Comment:
at this point, what does it mean if a `pIn` does not specify a taproot merkle root?

Reply:
^ Could add that to the comment here.